### PR TITLE
osl: 1.10.9 -> 1.11.17.0

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

fix build and update package. Closes https://github.com/NixOS/nixpkgs/pull/129248

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by viraptor** [comment]
Something's not right with boost right now:
```
libs/python/src/exec.cpp: In function 'boost::python::api::object boost::python::exec_file(const char*, boost::python::api::object, boost::python::api::object)':
libs/python/src/exec.cpp:109:14: error: '_Py_fopen' was not declared in this scope; did you mean '_Py_wfopen'?
  109 |   FILE *fs = _Py_fopen(f, "r");
      |              ^~~~~~~~~
      |              _Py_wfopen
```

**Comment by viraptor** [comment]
Result of `nixpkgs-review pr 182559` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>appleseed (python310Packages.appleseed)</li>
    <li>python39Packages.appleseed</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>blender</li>
    <li>opencolorio</li>
    <li>openimageio2</li>
    <li>osl</li>
    <li>python310Packages.bpycv</li>
    <li>python39Packages.bpycv</li>
  </ul>
</details>

**Comment by onny** [comment]
I would like to fix appleseed in a different PR. As far as I can tell it was broken before anyway

**Comment by risicle** [comment]
`nixpkgs-review` reveals no new failures, macos 10.15 & nixos x86_64

File: pkgs/development/compilers/osl/default.nix [hunk 1]
```diff
@@ -1,46 +1,76 @@
...
+, flex
+, bison
+, partio
+, pugixml
+, util-linux
+, python3 }:
+
+let
+
+  boost_static = boost165.override { enableStatic = true; };
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/osl/default.nix:21]: Like I wrote on the other PR. Can we move this to top-level, too?

File: pkgs/development/compilers/osl/default.nix [hunk 2]
```diff
@@ -1,46 +1,76 @@
...
+, openexr
+, openimageio2
+, llvm
+, boost165
+, flex
+, bison
+, partio
+, pugixml
+, util-linux
+, python3 }:
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/osl/default.nix:17]:
```suggestion
, python3
}:
```

File: pkgs/development/compilers/osl/default.nix [hunk 3]
```diff
@@ -1,46 +1,76 @@
...
+    "-DCMAKE_CXX_STANDARD=14"
   ];
 
   preConfigure = "patchShebangs src/liboslexec/serialize-bc.bash ";
 
-  nativeBuildInputs = [ cmake boost_static flex bison];
+  nativeBuildInputs = [
+    clang
+    cmake
+    boost_static
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/osl/default.nix:52]: boost in nativeBuildInputs? Don't we compile against it?

File: pkgs/development/compilers/osl/default.nix [hunk 4]
```diff
@@ -1,46 +1,76 @@
...
-     python3 # CMake doesn't check this?
+    libclang
+    zlib
+    openexr
+    openimageio2
+    llvm
+    partio
+    pugixml
+    util-linux # needed just for hexdump
+    python3
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/osl/default.nix:66]:
```suggestion
```
Isn't that already included with pybind11?

**Comment by onny** [comment]
Thank you @SuperSandro2000 for taking a look into it. Fixed the remaining issues :)

**Pull request merged**